tests/e2e/vmc_csi_deployments.go

```diff
@@ -0,0 +1,196 @@
+		defer cancel()
+		ginkgo.By("Modify Resource Quota")
+		svcClient, svNamespace := getSvcClientAndNamespace()
+		setResourceQuota(svcClient, svNamespace, defaultrqLimit)
+
+		if pvclaim != nil {
+			err := fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = e2eVSphere.waitForCNSVolumeToBeDeleted(volHandle)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+		if deployment != nil {
+			ginkgo.By("Delete Deployment")
+			err := client.AppsV1().Deployments(namespace).Delete(ctx, deployment.Name, metav1.DeleteOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		}
+
+	})
+
+	/*
+		Test performs following operations
+
+		Steps
+		1. Create Storage Class and PVC
+		2. Deploy nginx pods with volume
+		3. Update the nginx deployment pods, update the nginx image from k8s.gcr.io/nginx-slim:0.8 to k8s.gcr.io/nginx-slim:0.9
+		4. Wait for some time and verify the update is successful
+		5. Scale dowm the deployment to 0 replicas
+		6. Scale up the deployment to 1 replicas and verify all the pods should be up and running
+		7. Delete the deployments and Verify deployments are deleted successfully
+		8. Delete the PVC
+	*/
+	ginkgo.It("Deployments with volume provisioning", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		var err error
+		var lables = make(map[string]string)
+		lables["app"] = "nginx"
+		replica = 1
+
+		ginkgo.By("CNS_TEST: Running for GC setup")
+		scParameters[svStorageClassName] = storagePolicyName
+
+		ginkgo.By("Creating StorageClass for Deployment")
+		sc, pvclaim, err = createPVCAndStorageClass(client, namespace, nil, scParameters, diskSize, nil, "", false, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Creating PVC")
+		pvclaims = append(pvclaims, pvclaim)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		persistentvolumes, err := fpv.WaitForPVClaimBoundPhase(client, pvclaims, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		volHandle = persistentvolumes[0].Spec.CSI.VolumeHandle
+		gomega.Expect(volHandle).NotTo(gomega.BeEmpty())
+
+		defer func() {
+			err := fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = e2eVSphere.waitForCNSVolumeToBeDeleted(volHandle)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			pvclaim = nil
+		}()
+
+		ginkgo.By("Creating Deployments")
+		deployment, err := createDeployment(ctx, client, replica, lables, nil, namespace, pvclaims, "", false, nginxImage)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		deployment, err = client.AppsV1().Deployments(namespace).Get(ctx, deployment.Name, metav1.GetOptions{})
```

Comment:
why?

Reply:
I'm doing an update on the deployment so I am getting the deployment and modifying it

Reply:
there is no update before this call right?

Reply:
yes, without this get call I am getting this issue 
`Operation cannot be fulfilled on deployments.apps "deployment-18c50642-8199-4c64-8989-121748f53dee": the object has been modified; please apply your changes to the latest version and try again`